### ISSUE TITLE
iOS: Toggle buttons should change accessibility text when moving between states

### ISSUE THREAD

**Issue by vincode-io** [Ranchero-Software/NetNewsWire#1581]
#1
On the main screen of the app, the “Filter Feeds” button should offer more feedback on what it does and its current selection status.

Currently, the VoiceOver label for this button is the same regardless of its selection status.

Am I correct in believing that this button toggles between showing and hiding feeds which have unread articles?

If so, I would suggest that the button have the following two labels depending upon its selection status:

Unread Feeds
Selected - Unread Feeds

You might also use the VoiceOver ‘hints’ feature to provide more information on the function of this button. For example:

“show only feeds with unread articles”

It's worth noting that VoiceOver speaking of hints is something that users can disable, so we shouldn't rely solely on these to provide an indication of what UI elements are.

#2
When viewing a list of articles in a feed or folder, the “Filter Articles” button should offer more feedback on what it does and its current selection status.

For consistency, I would suggest that there are again two different (but similar to those use elsewhere) labels for this button:

Unread Articles
Selected - Unread Articles

As the function of this button should be clear from these VoiceOver labels, there is probably little value in also having VoiceOver hints (which in this case would have the negative effect of adding extra verbosity, in that VoiceOver would be speaking more than is necessary or desirable for the user experience).

#3
When viewing an article, the “Toggle Read” and “Toggle Starred” buttons need to indicate their current selection status.

Currently, both buttons use the same VoiceOver label for both selected and unselected. I would suggest that this be changed to:

Mark Article Read
Selected - Mark Article Read

Star Article
Selected - Star Article

Again, I would suggest that NoiceOver hints would only add unnecessary verbosity.

**Comment by vincode-io** [comment]
#4
The Article Extractor Button should use "Reader View", "Processing - Reader View", "Selected - Reader View", and "Error - Reader View" as accessiblity labels to match its button state.
